src/bin/tikv-server.rs

```diff
@@ -344,6 +344,28 @@ fn configure_grpc_poll_strategy() {
     }
 }
 
+fn check_and_persist_critical_config(config: &TiKvConfig, matches: &ArgMatches) {
+    if let Some(path) = matches.value_of("config") {
+        let cfg_file = Path::new(path);
+        if let Some(cfg_dir) = cfg_file.parent() {
+            // Check current critical configurations with last time, if there are some
+            // changes, user must guarantee relevant works have been done.
+            let last_cfg_file = auto_gen_cfg_file(cfg_dir.to_str().unwrap());
```

Comment:
Better not to convert `Path/PathBuf` with `&str` back and forth, it's more convenient to use `Path`, and you don't need to join the strings manually.